src/Symfony/Bundle/FrameworkBundle/FrameworkBundle.php

```diff
@@ -70,6 +71,10 @@ public function boot()
         if ($trustedHosts = $this->container->getParameter('kernel.trusted_hosts')) {
             Request::setTrustedHosts($trustedHosts);
         }
+
+        if ($this->container->hasParameter('form.allowed_methods')) {
+            FormConfigBuilder::setAllowedMethods($this->container->getParameter('form.allowed_methods'));
```

Comment:
I would suggest to try harder *not* using any global state for this feature.
Could be just an option for `FormConfigBuilder` maybe?

Reply:
`FormConfigBuilder` is not instantiated by the DI container, but by the `ResolvedFormType` class which is created by `ResolvedFormTypeFactory`, so the injection of a parameter is too cumbersome. I tried to create a TypeExtension to inject the parameter as an option, but it can be overridden on any form type (this means that we can pass method and allowed methods as options and it I don't think it makes sense).

Now i tried to inject the allowed methods directly in the form factory which is set into the `FormConfigBuilder`. What do you think about this solution?